src/main/java/uk/gov/hmcts/reform/ccd/documentam/service/impl/DocumentManagementServiceImpl.java

```diff
@@ -3,202 +3,120 @@
 import uk.gov.hmcts.reform.ccd.documentam.model.Document;
 import uk.gov.hmcts.reform.ccd.documentam.model.DocumentHashToken;
 import uk.gov.hmcts.reform.ccd.documentam.model.DocumentPermissions;
 import uk.gov.hmcts.reform.ccd.documentam.model.DocumentUpdate;
 import uk.gov.hmcts.reform.ccd.documentam.model.PatchDocumentResponse;
-import uk.gov.hmcts.reform.ccd.documentam.model.StoredDocumentHalResource;
 import uk.gov.hmcts.reform.ccd.documentam.model.UpdateDocumentsCommand;
-import uk.gov.hmcts.reform.ccd.documentam.model.UpdateTtlRequest;
-import uk.gov.hmcts.reform.ccd.documentam.model.UploadResponse;
 import uk.gov.hmcts.reform.ccd.documentam.model.enums.Permission;
-import uk.gov.hmcts.reform.ccd.documentam.service.CaseDataStoreService;
+import uk.gov.hmcts.reform.ccd.documentam.client.datastore.CaseDataStoreClient;
 import uk.gov.hmcts.reform.ccd.documentam.service.DocumentManagementService;
 import uk.gov.hmcts.reform.ccd.documentam.util.ApplicationUtils;
-import uk.gov.hmcts.reform.ccd.documentam.util.ResponseHelper;
 
 import java.time.ZoneId;
-import java.time.ZonedDateTime;
 import java.util.ArrayList;
 import java.util.Date;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.UUID;
 import java.util.stream.Collectors;
 
-import static org.springframework.http.HttpMethod.DELETE;
-import static org.springframework.http.HttpMethod.GET;
-import static org.springframework.http.HttpMethod.PATCH;
+import static java.util.Collections.emptyList;
+import static uk.gov.hmcts.reform.ccd.documentam.apihelper.Constants.CASE_ID;
 import static uk.gov.hmcts.reform.ccd.documentam.apihelper.Constants.CASE_TYPE_ID;
-import static uk.gov.hmcts.reform.ccd.documentam.apihelper.Constants.DM_DATE_TIME_FORMATTER;
+import static uk.gov.hmcts.reform.ccd.documentam.apihelper.Constants.DOCUMENT_METADATA_NOT_FOUND;
+import static uk.gov.hmcts.reform.ccd.documentam.apihelper.Constants.RESOURCE_NOT_FOUND;
 
 @Slf4j
 @Service
 public class DocumentManagementServiceImpl implements DocumentManagementService {
 
     private static final Date NULL_TTL = null;
 
-    private final RestTemplate restTemplate;
+    private final DocumentStoreClient documentStoreClient;
 
-    @Value("${documentStoreUrl}")
-    protected String documentURL;
-
-    @Value("${documentTtlInDays}")
-    protected int documentTtlInDays;
-
-    @Value("${idam.s2s-auth.totp_secret}")
-    protected String salt;
-
-    private final CaseDataStoreService caseDataStoreService;
-
-    @Value("${hash.check.enabled}")
-    private boolean hashCheckEnabled;
-
-    @Value("${bulkscan.exception.record.types}")
-    private List<String> bulkScanExceptionRecordTypes;
+    private final CaseDataStoreClient caseDataStoreClient;
 
     private final AuthorisedServices authorisedServices;
 
-    private static final HttpEntity<Object> NULL_REQUEST_ENTITY = null;
+    private final ApplicationParams applicationParams;
 
     @Autowired
```

Comment:
Appreciate its not your change, but `@Inject` and `@Autowired` both used to same effect in this repo - perhaps another ticket required to ensure consistency?